src/test/java/com/example/onlinebookstore/controller/CategoryControllerTest.java

```diff
@@ -0,0 +1,273 @@
+                        .getContentAsString(), CategoryResponseDto.class);
+
+        assertNotNull(actual);
+        assertNotNull(actual.id());
+        Assertions.assertTrue(EqualsBuilder.reflectionEquals(expected, actual,
+                "id"));
+    }
+
+    @SneakyThrows
+    @WithMockUser(username = "user")
+    @Test
+    @DisplayName("Get Category by valid id")
+    @Sql(scripts = "classpath:database/categories/add-category.sql")
+    void getCategoryById_ValidId_Success() {
+        long categoryId = 1L;
+
+        CategoryResponseDto expected = new CategoryResponseDto(
+                1L,
+                "Adults",
+                "Aboba 10 IQ"
+        );
+
+        MvcResult result = mockMvc.perform(get("/api/categories/" + categoryId)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        CategoryResponseDto actual = objectMapper.readValue(result.getResponse()
+                        .getContentAsString(), CategoryResponseDto.class);
+
+        assertNotNull(actual);
+        assertNotNull(actual.id());
+        Assertions.assertTrue(EqualsBuilder.reflectionEquals(expected, actual));
+    }
+
+    @SneakyThrows
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @DisplayName("Delete category by valid id")
+    @Sql(scripts = "classpath:database/categories/add-category.sql")
+    void deleteCategoryById_ValidId_Success() {
+        long categoryId = 1L;
+
+        mockMvc.perform(delete("/api/categories/" + categoryId))
+                .andExpect(status().isNoContent())
+                .andReturn();
+    }
+
+    @SneakyThrows
+    @WithMockUser(username = "user")
+    @Test
+    @DisplayName("Find all categories")
+    @Sql(scripts = "classpath:database/categories/add-three-categories.sql")
+    void findAllCategories_Valid_Success() {
+
+        CategoryResponseDto category1 = new CategoryResponseDto(
+                1L,
+                "Adults",
+                "Aboba 10 IQ"
+        );
+
+        CategoryResponseDto category2 = new CategoryResponseDto(
+                2L,
+                "children",
+                "Aboba 5 IQ"
+        );
+
+        CategoryResponseDto category3 = new CategoryResponseDto(
+                3L,
+                "third category",
+                "Aboba 1 IQ"
+        );
+
+        List<CategoryResponseDto> expected = List.of(category1, category2, category3);
+
+        MvcResult result = mockMvc.perform(get("/api/categories")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        String resultAsString = result.getResponse().getContentAsString();
+
+        List<CategoryResponseDto> actual =
+                objectMapper.readerForListOf(CategoryResponseDto.class)
+                        .readValue(resultAsString);
+
+        for (int i = 0; i < expected.size(); i++) {
```

Comment:
I would suggest using assertIterableEquals(expected, actual). In this case you can skip using for loop (but its up to you)